x-pack/plugins/fleet/server/routes/message_signing_service/handlers.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { TypeOf } from '@kbn/config-schema';
+
+import type { FleetRequestHandler } from '../../types';
+
+import { defaultFleetErrorHandler } from '../../errors';
+import { appContextService } from '../../services';
+import type { RotateKeyPairSchema } from '../../types/rest_spec/message_signing_service';
+
+export const rotateKeyPairHandler: FleetRequestHandler<
+  undefined,
+  TypeOf<typeof RotateKeyPairSchema.query>,
+  undefined
+> = async (_, __, response) => {
+  try {
+    await appContextService.getMessageSigningService()?.rotateKeyPair();
```

Comment:
@dasansol92 This would return `void | undefined`. I think we should check if the rotateKeyPair is actually a success before responding as `ok`. So maybe have `rotateKeyPair()` return a `boolean` instead so we can check that here.

Reply:
I guess if there is an error during `rotateKeyPair` it will throw an error that will be catch by the `try/catch`. Let me know if you think different, I'm ok adding extra checks if needed.

Reply:
Not an error but when the generate doesn't actually go through. I notice that the generateKeyPair is called conditionally in `rotateKeyPair()` so it might happen that delete doesn't happen and so the generate doesn't either. Another case is what happens when messageSigningService is not available. I'm thinking something like this instead.

```json5
try {
    const rotateKeyPairResponse = await appContextService
      .getMessageSigningService()
      ?.rotateKeyPair();

    if (!rotateKeyPairResponse) {
      return response.customError({
        statusCode: 500,
        body: {
          message: 'Failed to rotate key pair!',
        },
      });
    }
    return response.ok({
      body: {
        message: 'Key pair rotated successfully.',
      },
    });
  } catch (error) {
    return defaultFleetErrorHandler({ error, response });
  }

Reply:
Yeah, looks good, go for it!

Reply:
done 7bedeebd93768ca9a6a907564271acdbcdf9b4f2

Reply:
Can you update the unit test too? Thanks!

Reply:
On it 😄

Reply:
tests 4cb2d911093f864f3bafc8523dccd79372332eda